docs/library/bluetooth.rst

```diff
@@ -0,0 +1,306 @@
+:mod:`bluetooth` --- low-level Bluetooth
+========================================
+
+.. module:: bluetooth
+   :synopsis: Low-level Bluetooth radio functionality
+
+This module provides an interface to a Bluetooth controller on a board.
+Currently this supports Bluetooth Low Energy (BLE) in Central, Peripheral,
+Broadcaster, and Observer roles.
+
+This API is intended to match the low-level Bluetooth protocol and provide
+building-blocks for higher-level abstractions such as specific device types.
+
+class BLE
+---------
+
+Constructor
+-----------
+
+.. class:: BLE()
+
+    Returns the singleton BLE object.
+
+Configuration
+-------------
+
+.. method:: BLE.active([active])
+
+    Optionally changes the active state of the BLE radio, and returns the
+    current state.
+
+    The radio must be made active before using any other methods on this class.
+
+.. method:: BLE.config(name)
+
+    Queries a configuration value by *name*. Currently supported values are:
+
+    - ``'mac'``: Returns the device MAC address. If a device has a fixed address
+      (e.g. PYBD) then it will be returned. Otherwise (e.g. ESP32) a random
+      address will be generated when the BLE interface is made active.
+
+Event Handling
+--------------
+
+.. method:: BLE.irq(handler, [trigger=0xffff])
```

Comment:
The square brackets around the trigger arg are no longer needed since it specifies a default